File: frontend/src/lib/stats.ts
/** 消费分析聚合 API（服务端 SQL 聚合，前端只负责画图）。 */
import { api } from './auth';
import { DateRange } from './dateFilter';

export interface StatBucket {
  amount: number;
  count: number;
}
export interface CategoryStat {
  category: string;
  amount: number;
  count: number;
}
export interface MonthStat {
  month: string; // YYYY-MM
  amount: number;
  count: number;
}
export interface Stats {
  total: number;
  count: number;
  by_reimbursement: Record<string, StatBucket>;
  by_category: CategoryStat[];
  by_month: MonthStat[];
}

export function getStats(range: DateRange): Promise<Stats> {
  const p = new URLSearchParams();
  if (range.from) p.set('date_from', range.from);
  if (range.to) p.set('date_to', range.to);
  const q = p.toString();
  return api.get<Stats>(`/invoices/stats${q ? `?${q}` : ''}`);
}

File: backend/app/schemas/stats.py
"""消费分析聚合（服务端 SQL group by，前端只画图不再全量 reduce）。"""

from pydantic import BaseModel


class StatBucket(BaseModel):
    amount: float
    count: int


class CategoryStat(BaseModel):
    category: str
    amount: float
    count: int


class MonthStat(BaseModel):
    month: str  # YYYY-MM
    amount: float
    count: int


class StatsOut(BaseModel):
    total: float
    count: int
    # 报销三态金额/张数：键为 unreimbursed/submitted/reimbursed，三态恒全（缺省为 0）
    by_reimbursement: dict[str, StatBucket]
    by_category: list[CategoryStat]  # 按金额降序
    by_month: list[MonthStat]  # 连续月份（缺月补 0），旧→新
